src/components/Admin/EditOpportunity.tsx

```diff
@@ -0,0 +1,158 @@
+import ProfileForm, { ProfileFormValuesType } from '../ProfileForm/ProfileForm';
+import { CreateContextInput, UpdateContextInput, UpdateReferenceInput } from '../../types/graphql-schema';
+import Typography from '../core/Typography';
+import Button from '../core/Button';
+import Loading from '../core/Loading';
+import FormMode from './FormMode';
+
+interface Params {
+  ecoverseId?: string;
+  challengeId?: string;
+  opportunityId?: string;
+}
+
+interface Props {
+  mode: FormMode;
+  paths: Path[];
+  title: string;
+}
+
+const EditOpportunity: FC<Props> = ({ paths, mode, title }) => {
+  const notify = useNotification();
+  const handleError = useApolloErrorHandler();
+  const onSuccess = (message: string) => notify(message, 'success');
+
+  const {
+    ecoverseId = '',
+    opportunityId: opportunityNameId = '',
+    challengeId: challengeNameId = '',
+  } = useParams<Params>();
+
+  const [createOpportunity, { loading: isCreating }] = useCreateOpportunityMutation({
+    refetchQueries: [refetchOpportunitiesQuery({ ecoverseId, challengeId: challengeNameId })],
+    awaitRefetchQueries: true,
+    onCompleted: () => onSuccess('Successfully created'),
+    onError: handleError,
+  });
+  const [updateOpportunity, { loading: isUpdating }] = useUpdateOpportunityMutation({
+    onCompleted: () => onSuccess('Successfully updated'),
+    onError: handleError,
+    refetchQueries: [refetchOpportunityProfileInfoQuery({ ecoverseId, opportunityId: opportunityNameId })],
+    awaitRefetchQueries: true,
+  });
+
+  const { data: opportunityProfile } = useOpportunityProfileInfoQuery({
+    variables: { ecoverseId: ecoverseId, opportunityId: opportunityNameId },
+    skip: mode === FormMode.create,
+  });
+
+  const opportunity = opportunityProfile?.ecoverse?.opportunity;
+  const opportunityId = useMemo(() => opportunity?.id || '', [opportunity]);
+
+  const isLoading = isCreating || isUpdating;
+
+  const currentPaths = useMemo(() => [...paths, { name: opportunity?.displayName || 'new', real: false }], [
+    paths,
+    opportunity,
+  ]);
+  useUpdateNavigation({ currentPaths });
+
+  const onSubmit = async (values: ProfileFormValuesType) => {
+    const { name, nameID, background, impact, tagline, vision, who, references, visual, tagsets } = values;
+
+    const updatedRefs: UpdateReferenceInput[] = references.map<UpdateReferenceInput>(r => ({
+      ID: r.id,
+      description: r.description,
+      name: r.name,
+      uri: r.uri,
+    }));
+
+    const updateContext: UpdateContextInput = {
+      background: background,
+      impact: impact,
+      references: updatedRefs,
+      tagline: tagline,
+      vision: vision,
+      visual: visual,
+      who: who,
+    };
+    const createContext: CreateContextInput = {
+      background: background,
+      impact: impact,
+      references: references,
+      tagline: tagline,
+      vision: vision,
+      visual: visual,
+      who: who,
+    };
```

Comment:
Could be extracted in utils/helper functions and reused on several different places.